zookeeper-server/src/main/java/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -588,9 +589,7 @@ public void processRequest(Request request) {
             StringBuilder sb = new StringBuilder();
             byte[] payload = request.readRequestBytes();
             if (payload != null) {
-                for (byte b : payload) {
-                    sb.append(String.format("%02x", (0xff & b)));
-                }
+                HexUtil.encodeHex(payload, sb);
```

Comment:
- It could not be better if you can use the `JMH` framework to test this new method comparing to the old one, convincing us the new one is more efficient.
- BTW, `PrepRequestProcessor` also has this code snippet